code/game/machinery/computer/telecrystalconsoles.dm

```diff
@@ -58,15 +58,25 @@ GLOBAL_LIST_INIT(possible_uplinker_IDs, list("Alfa","Bravo","Charlie","Delta","E
 
 /obj/machinery/computer/telecrystals/uplinker/proc/donateTC(amt, addLog = 1)
 	if(uplinkholder && linkedboss)
-		if(amt <= uplinkholder.hidden_uplink.telecrystals)
+		if(amt < 0)
+			linkedboss.storedcrystals += uplinkholder.hidden_uplink.telecrystals
+			if(addLog)
+				linkedboss.logTransfer("[src] donated [uplinkholder.hidden_uplink.telecrystals] telecrystals to [linkedboss].")
+			uplinkholder.hidden_uplink.telecrystals = 0
+		else if(amt <= uplinkholder.hidden_uplink.telecrystals)
 			uplinkholder.hidden_uplink.telecrystals -= amt
 			linkedboss.storedcrystals += amt
 			if(addLog)
 				linkedboss.logTransfer("[src] donated [amt] telecrystals to [linkedboss].")
 
 /obj/machinery/computer/telecrystals/uplinker/proc/giveTC(amt, addLog = 1)

```

Comment:
it's the numbers from the nuke op telecrystal transfer consoles, if it's getting numbers besides the numbers either of them have on hand then something is very wrong elsewhere